src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -337,6 +339,23 @@ public static void clearSnapshot(String snapshotName, String keyspace)
         return list;
     }
 
+    public List<SnapshotDetails> getSnapshotDetails() {
+        List<SnapshotDetails> list = new ArrayList<>();
+        for (ColumnFamilyStore cfStore : getColumnFamilyStores())
+        {
+            Directories dirs = cfStore.getDirectories();
+            for (Map.Entry<String, Directories.SnapshotSizeDetails> snapshotDetail : cfStore.getSnapshotDetails().entrySet())
+            {
+                String snapshotName = snapshotDetail.getKey();
+                File manifest = dirs.getSnapshotManifestFile(snapshotName);
```

Comment:
@fibersel as discussed in the ticket already, I think this call is not deterministic and it should be addressed. We should have a deterministic wait to get a manifest for particular SnapshotDetails.